### PR TITLE
[TW-1540] Refresh "Design your API in Postman using the API Builder"

### PR DESCRIPTION
**Pull request by akinard-postman**
Refresh of the `/docs/designing-and-developing-your-api/the-api-workflow.md` article. As part of this update I:
- Created a *Contents* header with the contents of the page.
- Renamed "High-level steps" to "Designing and developing your API" for clarity of this section and its purpose.
- Moved the callout box near the top since the information is pretty relevant and having it at the bottom of the page did not seem like the best place for it.
- Added some linking out to other articles.
- Minor rewording and Vale error clean up.

### CODE REVIEW

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md [hunk 1]
```diff
@@ -20,38 +20,43 @@ contextual_links:
...
+- [Developing an API](/docs/designing-and-developing-your-api/developing-an-api/defining-an-api/) - Build out your API by [editing your API definition](/docs/designing-and-developing-your-api/developing-an-api/defining-an-api/#editing-an-api-definition-file) and [adding collections](/docs/designing-and-developing-your-api/developing-an-api/adding-api-elements/). You can also [generate server-side code](/docs/designing-and-developing-your-api/developing-an-api/generating-server-code/) from your OpenAPI 3 specification and [validate your API definition](/docs/designing-and-developing-your-api/developing-an-api/validating-elements-against-schema/) and linked elements.
+- [Testing an API](/docs/designing-and-developing-your-api/testing-an-api/) - [Add tests](/docs/designing-and-developing-your-api/testing-an-api/#adding-tests) to your API, and use [CI integration](/docs/designing-and-developing-your-api/testing-an-api/#adding-ci-integration) to build and automate testing.
```
Comment (mark-anthony-postman) [inline src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:41]:
```suggestion
- [Testing an API](/docs/designing-and-developing-your-api/testing-an-api/) - [Add tests](/docs/designing-and-developing-your-api/testing-an-api/#adding-api-tests) to your API, and use [CI integration](/docs/designing-and-developing-your-api/testing-an-api/#adding-ci-integration) to build and automate testing.
```

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md [hunk 2]
```diff
@@ -20,38 +20,43 @@ contextual_links:
...
 
-When you select an API in the sidebar, the workbench displays an overview of the API. The overview displays a description of the API, tags associated with the API ([Enterprise plans only](https://www.postman.com/pricing)), linked collections, and published versions. The overview also provides quick access to tasks such as connecting to a repository, adding tests, observing API performance, and viewing deployments.
+When you select an API in the sidebar, the workbench displays an overview of the API. The overview displays a description of the API, tags associated with the API ([Enterprise plans](https://www.postman.com/pricing)), any linked collections, and the API's published versions. The overview also provides quick access to tasks, such as connecting to a Git repository, adding tests, observing API performance, and viewing deployments.
 
-<img src="https://assets.postman.com/postman-docs/v10/api-builder-overview-v10-3.jpg" alt="API Builder overview" />
+![API Builder overview](https://assets.postman.com/postman-docs/v10/api-builder-overview-v10-21-1.jpg)
```
Comment (mark-anthony-postman) [inline src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:56]: Mind re-taking this screenshot to capture the new tab UI (orange highlight is now on bottom of tab)

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md [hunk 3]
```diff
@@ -20,38 +20,43 @@ contextual_links:
...
+Postman supports [API-first development](https://www.postman.com/api-first/) with the API Builder. You can use the API Builder to [design your API](https://www.postman.com/api-platform/api-design/) in Postman before [developing](/docs/designing-and-developing-your-api/developing-an-api/develop-an-api/) it. Your API definition can then act as the single source of truth for your API project.
 
-You can connect various elements of your [API development lifecycle](https://www.postman.com/api-platform/api-lifecycle/) to your API definition, such as collections, documentation, and tests. You can also sync your API in Postman with a Git repository.
+With the API Builder, you can connect various Postman elements of your [API development lifecycle](https://www.postman.com/api-platform/api-lifecycle/) to your API definition, such as collections, documentation, and tests. You can also [integrate your API](/docs/designing-and-developing-your-api/versioning-an-api/versioning-an-api-overview/) in Postman with a Git repository, such as GitHub, GitLab, or Azure DevOps.
```
Comment (mark-anthony-postman) [inline src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:25]:
```suggestion
With the API Builder, you can connect various Postman elements of your [API development lifecycle](https://www.postman.com/api-platform/api-lifecycle/) to your API definition, such as collections, documentation, and tests. You can also [integrate your API](/docs/designing-and-developing-your-api/versioning-an-api/versioning-an-api-overview/) in Postman with a Git repository, such as GitHub, GitLab, Bitbucket, or Azure DevOps.
```

Comment (mark-anthony-postman) [reply src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:25]: I worried Bitbucket would feel left out 😀

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md (hunk 1 above)
Comment (MatthewPinsker-postman) [inline src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:41]:
```suggestion
- [Testing an API](/docs/designing-and-developing-your-api/testing-an-api/) - [Add tests](/docs/designing-and-developing-your-api/testing-an-api/#adding-api-tests) to your API, and use [CI integration](/docs/designing-and-developing-your-api/testing-an-api/#adding-ci-integration) to build and automate testing.
```

**Review by mark-anthony-postman — APPROVED** [review]
Nice refresh! 🌟 Don't forget to update the date 😀

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md (hunk 1 above)
Comment (mark-anthony-postman) [reply src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:41]: Aw Matthew beat me to it 😆

**Review by MatthewPinsker-postman — APPROVED** [review]
LGTM!

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md (hunk 1 above)
Comment (akinard-postman) [reply src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:41]: Thanks for catching that!

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md (hunk 2 above)
Comment (akinard-postman) [reply src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:56]: Is this live? I was checking in **pm-tech-writers** (it's where I did the mock ups) and that wasn't updated to the new style.

Comment (mark-anthony-postman) [reply src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:56]: Yes, it's live. I can see it in prod for Postman team, and on my free account

Comment (akinard-postman) [reply src/pages/docs/designing-and-developing-your-api/the-api-workflow.md]: Should be fixed in 8ad164400b11d36f269205ffc8906e5d5483ab7d

File: src/pages/docs/designing-and-developing-your-api/the-api-workflow.md [hunk 4]
```diff
@@ -20,38 +20,43 @@ contextual_links:
     blog_tag: "api-first"
 ---
```
Comment (jkonrath-postman) [inline src/pages/docs/designing-and-developing-your-api/the-api-workflow.md:21]: Change the `updated` date to 2024-01-18

**Review by jkonrath-postman — APPROVED** [review]
LGTM

**Pull request merged**